lib/plugins/plan/plan.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const validate = require('../lib/validate');
+
+class Plan {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+
+    Object.assign(
+      this,
+      validate
+    );
+
+    this.commands = {
+      plan: {
+        usage: 'Plan a Serverless service',
+        lifecycleEvents: [
+          'deprecated#cleanup->package:cleanup',
+          'deprecated#initialize->package:initialize',
+          'deprecated#setupProviderConfiguration->package:setupProviderConfiguration',
+          'deprecated#createDeploymentArtifacts->package:createDeploymentArtifacts',
+          'deprecated#compileFunctions->package:compileFunctions',
+          'deprecated#compileEvents->package:compileEvents',
+          'plan',
+          'finalize',
+        ],
+        options: {
+          stage: {
+            usage: 'Stage of the service',
+            shortcut: 's',
+          },
+        },
+        commands: {
+          plan: {
+            usage: 'Deploy a single function from the service',
+            lifecycleEvents: [
+              'initialize',
+              'packageFunction',
+              'plan',
+            ],
+          },
+        },
+      },
+    };
+
+    this.hooks = {
+      'before:plan:plan': () => BbPromise.bind(this)
+        .then(this.validate)
+        .then(() => {
```

Comment:
for consistency with e.g. the line above, you could make this a named (and tested) function, rather than anonymous